### PR TITLE
Refs #24121 -- Added __repr__() to PermWrapper.

### PR DESCRIPTION
**Pull request by abhiabhi94**

### CODE REVIEW

File: django/contrib/auth/context_processors.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ class PermWrapper:
     def __init__(self, user):
         self.user = user
 
+    def __repr__(self):
+        return f'{self.__class__.__qualname__} user={self.user!r}'
```
Comment (felixxm) [inline django/contrib/auth/context_processors.py:29]: Kwargs notation is not necessary. Also parentheses are missing:
```suggestion
        return f'{self.__class__.__name__}({self.user!r})'
```

File: tests/auth_tests/test_context_processors.py [hunk 2]
```diff
@@ -33,6 +33,10 @@ def __eq__(self, other):
             self.eq_calls += 1
             return False
 
+    def test_repr(self):
+        perms = PermWrapper(MockUser())
+        self.assertEqual(repr(perms), f'{perms.__class__.__qualname__} user={perms.user!r}')
```
Comment (felixxm) [inline tests/auth_tests/test_context_processors.py:38]: Please compare to the expected value, e.g. 'PermWrapper(...)'.

File: django/contrib/auth/context_processors.py (hunk 1 above)
Comment (abhiabhi94) [reply django/contrib/auth/context_processors.py:29]: isn't `qualname` better than `name`?

Comment (felixxm) [reply django/contrib/auth/context_processors.py:29]: It's a top-level class internal class so it doesn't matter.

File: tests/auth_tests/test_context_processors.py (hunk 2 above)
Comment (abhiabhi94) [reply tests/auth_tests/test_context_processors.py]: I have updated the values as you asked for, but I'm curious to know if checking it be to `PermWrapper` is better than `perms.__class__.__name__`? Wouldn't we have to change the tests, if by any chance we change the class name?

I'm just asking this to know which is kind of a better design for a test. I tend to use the non-constant form till now in my tests.

Comment (felixxm) [reply tests/auth_tests/test_context_processors.py]: > Wouldn't we have to change the tests, if by any chance we change the class name?

Yes and it's expected, we will be aware which tests are affected by our change to the base class. IMO it's also more readable.

**Comment by abhiabhi94** [comment]
Thanks @felixxm for the review.

**Pull request merged**